src/main/java/com/google/sps/data/Room.java

```diff
@@ -106,13 +108,25 @@ public static Room fromKey(Key roomKey) {
         return null;
     }
 
-    //Adds a message to the room
-    public void addMessage(Message message){
-        if(this.messages.size() < 10) {
-            this.messages.add(message);
-        } else {
-            this.messages.remove(0);
-            this.messages.add(message);
+    // Helper addMessage function to manipulate the list of messages
+    private void addMessage(Message msg){
+        if(messages.size() >= 10) {
+            messages.remove(0);
+        }
+        messages.add(msg);
+    }
+
+    // Manipulates the messages property of the room given a room key and a new message to be added
+    public static void addMessagesFromKey(Key roomKey, Message chatMessage) {
```

Comment:
Don't add a new static function to interface with the Room object. Instead have the servlet call a getRoomByRoomId function. Then perform the operations there.

The datastore.put should be kept here, if there is no function to do so. Create one.

Reply:
@dangelyjeanelle made this edit, and has the answers regarding this request

Reply:
In order to update the room entity, I must set the properties that need to be changed to the new values using the datastore entity. The getRoomById function returns a Room object with no reference to the actual datastore entity, for which the code would result in a new room entity every time the messages are updated.

Reply:
@KillianLeC